src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/PremisesController.kt

```diff
@@ -267,17 +267,11 @@ class PremisesController(
 
     return ResponseEntity.ok(
       premises.bookings.mapNotNull {
-        val personDetails = getPersonDetailsForCrn(log, it.crn, user.deliusUsername, offenderService, user.hasQualification(UserQualification.LAO))
+        val personInfo = offenderService.getInfoForPerson(it.crn, user.deliusUsername, user.hasQualification(UserQualification.LAO))
 
-        if (personDetails == null) {
-          log.warn("Unable to get Person via crn: ${it.crn}")
-          return@mapNotNull null
-        }
-
-        val (offenderDetails, inmateDetails) = personDetails
+        if (personInfo !is PersonInfoResult.Success) return@mapNotNull null
```

Comment:
When looking at a bedspace to make a new booking I think it's very important for the user to be able to see basic information about all bookings on that bed and in the rest of the property. 

By omitting this data for any reason we introduce the risk that a user makes an incompatible/unsafe booking because they incorrectly determine that there were no other bookings. When in fact we've hidden it from them because of an LAO or perhaps a connection problem to Delius.

I think we need to change this to always return a booking either in full (if the user has access) or the minimal and safe amount of info: CRN, Arrival date, Departure date, booking status.

Reply:
Sorry for the late response to this one, only just spotted the comment.  In the case where the person the Booking is for is LAO and the user doesn't pass the LAO check for them or have the LAO qualification then `personInfo` here will come from this return statement in OffenderService::getInfoForPerson: https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/804/files#diff-4a8c3767d589a6283081bfd38ef72be5ebc2dd5ec9d401127ded27adb5ec015cR429

This will be a `PersonInfoResult.Success.Restricted` which just contains the CRN and NOMS number - which I think is in line with what you're asking to happen?

There is the scenario where one of the upstream calls fails (e.g. the call to check the LAO access or get Offender info) in which case the Booking will just not be shown in the list which isn't great - do you think the preferable behaviour there would be to 500 rather than present partial data?

Reply:
> There is the scenario where one of the upstream calls fails (e.g. the call to check the LAO access or get Offender info) in which case the Booking will just not be shown in the list which isn't great

This is my concern 👍 I do think it would be better to fail safely with a 500 rather than let a user make a new booking without a full picture of nearby bookings. If one of the missing bookings represents a person in residence that could cause a risk for a new booking then we have a safety problem.